### PR TITLE
Clean talk project list

### LINKED ISSUE #762: Talk sidebar needs purging
the Talk sidebar on the "Talk" page from Zooniverse home has a list of random projects in it;
this needs to be purged and updated with all relevant projects.

### PR DESCRIPTION
**Pull request by chrissnyder**
- Closes #762

### CODE REVIEW

**Comment by aweiksnar** [comment]
Looks like a better query for the sidebar list, but #goToProjectTalk isn't changing projects for me on this branch. Is it changing projects for you?

**Comment by chrissnyder** [comment]
Yea, no idea why it was working for me earlier. `login` definitely doesn't exist for included `owner` within a   project resource. Had to add the call back in.

**Comment by edpaget** [comment]
@chrissnyder when you include the resource it's not saved at the links property. Doing `project = apiClient.type('projects').get('1', include: 'owner')` will cache the owner so a subsequent `project.get 'owner'` won't hit the API and will return a promise with the previously included owner resource.

File: app/talk/lib/project-linker.cjsx [hunk 1]
```diff
@@ -18,21 +18,31 @@ module?.exports = React.createClass
     nextState.projects isnt @state.projects
 
   goToProjectTalk: (projectId) ->
-    apiClient.type('projects').get(projectId.toString()).then (project) =>
-      project.get('owner').then (owner) =>
+    selectedProject = null
+
+    for project in @state.projects
+      if project.id is projectId
+        selectedProject = project
```
Comment (aweiksnar) [inline app/talk/lib/project-linker.cjsx]: I think selectedProject is set but never used, unless I'm reading this wrong

**Comment by chrissnyder** [comment]
@edpaget That would do it. Query adjusted.

**Comment by vrooje** [comment]
Will this also sort the projects in the sidebar by something obvious like project title? Currently it seems random.

**Comment by chrissnyder** [comment]
Still random. The front end doesn't do any sorting; it just accepts however the API sends it.

**Comment by edpaget** [comment]
They're sorted by most recently updated.

**Pull request merged**